running pytest on fresh installation

I'm trying to set up board on a fresh ubuntu server. I managed to do this a month ago but now I can't get it work.

Here is what I did to set up the test db:

```
conda install postgresql
initdb postgres_dbs
pg_ctl -D postgres_dbs -l logfile start
createuser --pwprompt mrramp
createdb --owner=mrramp databoard_test
```

Then I edited db_engine.yml. 

```
db_owner: mrramp # to find out your db_owner type psql -l in the terminal
db_cluster_name: databoard_test # name of the engine database you have created
```

First `db_cluster_name` must be the name of the db `databoard_test`, not the name of the cluster (`postgres_dbs`; I got `FATAL:  database "postgres_dbs" does not exist` when I did this). Second, a slight annoyance is that now we cannot run pytest in `ramp-board/ramp-database`, we need to cd into `ramp-board`. Third, I can't get it run, it seems that the test script wants to recreate the db user. I'm not a big postgres expert so I don't want to do trial and error on this, maybe @maikia or @rth can help?

Here is the trace.

```
(base) ubuntu@ip-172-31-10-68:~/RAMP/ramp-board$ pytest -vsl ramp-database/ramp_database/model/tests/test_event.py::test_event_model_property
========================================================================= test session starts =========================================================================
platform linux -- Python 3.7.4, pytest-5.2.1, py-1.8.0, pluggy-0.13.0 -- /home/ubuntu/anaconda3/bin/python
cachedir: .pytest_cache
rootdir: /home/ubuntu/RAMP/ramp-board/ramp-database
plugins: doctestplus-0.4.0, arraydiff-0.3, remotedata-0.3.2, openfiles-0.4.0
collected 1 item                                                                                                                                                      

ramp-database/ramp_database/model/tests/test_event.py::test_event_model_property user mrramp already exists
ERROR

=============================================================================== ERRORS ================================================================================
_____________________________________________________________ ERROR at setup of test_event_model_property _____________________________________________________________

self = <sqlalchemy.engine.base.Connection object at 0x7ff448264550>, dialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7ff4481fc410>
constructor = <bound method DefaultExecutionContext._init_statement of <class 'sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2'>>
statement = "CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER"
parameters = {}
args = ("CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER", [])
conn = <sqlalchemy.pool.base._ConnectionFairy object at 0x7ff447cf4a90>
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7ff447d016d0>

    def _execute_context(
        self, dialect, constructor, statement, parameters, *args
    ):
        """Create an :class:`.ExecutionContext` and execute, returning
        a :class:`.ResultProxy`."""
    
        try:
            try:
                conn = self.__connection
            except AttributeError:
                # escape "except AttributeError" before revalidating
                # to prevent misleading stacktraces in Py3K
                conn = None
            if conn is None:
                conn = self._revalidate_connection()
    
            context = constructor(dialect, self, conn, *args)
        except BaseException as e:
            self._handle_dbapi_exception(
                e, util.text_type(statement), parameters, None, None
            )
    
        if context.compiled:
            context.pre_exec()
    
        cursor, statement, parameters = (
            context.cursor,
            context.statement,
            context.parameters,
        )
    
        if not context.executemany:
            parameters = parameters[0]
    
        if self._has_events or self.engine._has_events:
            for fn in self.dispatch.before_cursor_execute:
                statement, parameters = fn(
                    self,
                    cursor,
                    statement,
                    parameters,
                    context,
                    context.executemany,
                )
    
        if self._echo:
            self.engine.logger.info(statement)
            if not self.engine.hide_parameters:
                self.engine.logger.info(
                    "%r", sql_util._repr_params(parameters, batches=10)
                )
            else:
                self.engine.logger.info(
                    "[SQL parameters hidden due to hide_parameters=True]"
                )
    
        evt_handled = False
        try:
            if context.executemany:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_executemany:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_executemany(
                        cursor, statement, parameters, context
                    )
            elif not parameters and context.no_parameters:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute_no_params:
                        if fn(cursor, statement, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute_no_params(
                        cursor, statement, context
                    )
            else:
                if self.dialect._has_events:
                    for fn in self.dialect.dispatch.do_execute:
                        if fn(cursor, statement, parameters, context):
                            evt_handled = True
                            break
                if not evt_handled:
                    self.dialect.do_execute(
>                       cursor, statement, parameters, context
                    )

args       = ("CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER", [])
conn       = <sqlalchemy.pool.base._ConnectionFairy object at 0x7ff447cf4a90>
constructor = <bound method DefaultExecutionContext._init_statement of <class 'sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2'>>
context    = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7ff447d016d0>
cursor     = <cursor object at 0x7ff4482f3750; closed: -1>
dialect    = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7ff4481fc410>
evt_handled = False
parameters = {}
self       = <sqlalchemy.engine.base.Connection object at 0x7ff448264550>
statement  = "CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER"

../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/base.py:1249: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7ff4481fc410>, cursor = <cursor object at 0x7ff4482f3750; closed: -1>
statement = "CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER"
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7ff447d016d0>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       psycopg2.errors.InsufficientPrivilege: permission denied to create role

context    = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7ff447d016d0>
cursor     = <cursor object at 0x7ff4482f3750; closed: -1>
parameters = {}
self       = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7ff4481fc410>
statement  = "CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER"

../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/default.py:580: InsufficientPrivilege

The above exception was the direct cause of the following exception:

    @pytest.fixture(scope='session')
    def database_connection():
        """
        Create a Postgres database for the tests,
        and drop it when the tests are done.
        """
        config = safe_load(open("db_engine.yml"))
        dbowner = config.get('db_owner')
        dbcluster = config.get('db_cluster_name')
        engine = create_engine(f'postgresql://{dbowner}:@localhost/{dbcluster}',
                               isolation_level='AUTOCOMMIT')
    
        connection = engine.connect()
    
        database_config = read_config(database_config_template())
        username = database_config['sqlalchemy']['username']
        database_name = database_config['sqlalchemy']['database']
        try:
            connection.execute(f"""CREATE USER {username}
                                  WITH PASSWORD '{username}';
>                                 ALTER USER {username} WITH SUPERUSER""")

config     = {'db_cluster_name': 'databoard_test', 'db_owner': 'mrramp'}
connection = <sqlalchemy.engine.base.Connection object at 0x7ff448264550>
database_config = {'flask': {'debug': True, 'mail_default_sender': ['RAMP admin', 'rampmailer@localhost.com'], 'mail_port': 8025, 'mail_...qlalchemy': {'database': 'databoard_test', 'drivername': 'postgresql', 'host': 'localhost', 'password': 'mrramp', ...}}
database_name = 'databoard_test'
dbcluster  = 'databoard_test'
dbowner    = 'mrramp'
engine     = Engine(postgresql://mrramp:***@localhost/databoard_test)
username   = 'mrramp'

conftest.py:33: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/base.py:982: in execute
    return self._execute_text(object_, multiparams, params)
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/base.py:1155: in _execute_text
    parameters,
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/base.py:1253: in _execute_context
    e, statement, parameters, cursor, context
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/base.py:1473: in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/util/compat.py:398: in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/util/compat.py:152: in reraise
    raise value.with_traceback(tb)
../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/base.py:1249: in _execute_context
    cursor, statement, parameters, context
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7ff4481fc410>, cursor = <cursor object at 0x7ff4482f3750; closed: -1>
statement = "CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER"
parameters = {}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7ff447d016d0>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InsufficientPrivilege) permission denied to create role
E       
E       [SQL: CREATE USER mrramp 
E                                     WITH PASSWORD 'mrramp';
E                                     ALTER USER mrramp WITH SUPERUSER]
E       (Background on this error at: http://sqlalche.me/e/f405)

context    = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7ff447d016d0>
cursor     = <cursor object at 0x7ff4482f3750; closed: -1>
parameters = {}
self       = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7ff4481fc410>
statement  = "CREATE USER mrramp \n                              WITH PASSWORD 'mrramp';\n                              ALTER USER mrramp WITH SUPERUSER"

../../anaconda3/lib/python3.7/site-packages/sqlalchemy/engine/default.py:580: ProgrammingError
========================================================================== warnings summary ===========================================================================
/home/ubuntu/anaconda3/lib/python3.7/site-packages/nose/importer.py:12
  /home/ubuntu/anaconda3/lib/python3.7/site-packages/nose/importer.py:12: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import find_module, load_module, acquire_lock, release_lock

/home/ubuntu/RAMP/ramp-workflow/rampwf/utils/submission.py:7
  /home/ubuntu/RAMP/ramp-workflow/rampwf/utils/submission.py:7: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Iterable

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================================================================== 2 warnings, 1 error in 1.45s =====================================================================
```

--- a/conftest.py
+++ b/conftest.py
@@ -18,8 +18,8 @@ def database_connection():
     """
     config = safe_load(open("db_engine.yml"))
     dbowner = config.get('db_owner')
-    dbcluster = config.get('db_cluster_name')
-    engine = create_engine(f'postgresql://{dbowner}:@localhost/{dbcluster}',
+
+    engine = create_engine(f'postgresql://{dbowner}:@localhost/postgres',
                            isolation_level='AUTOCOMMIT')
 
     connection = engine.connect()

--- a/db_engine.yml
+++ b/db_engine.yml
@@ -1,2 +1,2 @@
-db_owner: mtelencz # to find out your db_owner type psql -l in the terminal
+db_owner: postgres # to find out your db_owner type psql -l in the terminal
                    # and check who is the owner of your postgres database 

--- a/db_engine.yml
+++ b/db_engine.yml
@@ -1,2 +1,2 @@
-db_owner: postgres # to find out your db_owner type psql -l in the terminal
-db_cluster_name: postgres # name of the engine database you have created
+db_owner: mtelencz # to find out your db_owner type psql -l in the terminal
+                   # and check who is the owner of your postgres database 

--- a/doc/contribute.rst
+++ b/doc/contribute.rst
@@ -23,10 +23,11 @@ that you use the ``make inplace`` option to install RAMP in developer mode.
 
 Prepare database engine
 -----------------------
-Testing RAMP require a database, you need to create a database similarly to
-:ref:`create database <set_database>`
+Testing RAMP requires a database cluster, you need to create it similarly as 
+described in :ref:`create database <set_database>` section. 
 
-In short, create the ``Postgres database cluster`` using the command::
+If you haven't done so already create the ``Postgres database cluster`` 
+using the command::
 
     ~ $ initdb postgres_dbs
 
@@ -34,25 +35,24 @@ and then start it with::
 
     ~ $ pg_ctl -D postgres_dbs -l logfile start
 
-However, RAMP tests do not know which database engine you are using for 
-your ramp development and therefore you need to inform RAMP tests about this
-To do this, open the ``db_engine.yml`` file located in the ``ramp-board`` directory. 
+Within your database cluster `postgres` database is created automatically.
+`pytest` will use it to make and then drop the test engine. But it needs
+to know who is the owner of your `postgres` database and therefore you 
+need to inform RAMP tests about this.
+To do this, open the ``db_engine.yml`` file located in the ``ramp-board`` 
+directory. 
 It should look as follows::
 
     db_owner: postgres
-    db_cluster_name: postgres
 
-You need to change <postgres> to your local settings.
+You need to change <postgres> to the owner of the `postgres` database. 
 
-``db_owner`` is the owner of your database. If you don't know who is 
-the owner of your database you can find it out by typing in your terminal::
+If you don't know who is the owner of your `postgres` database you can 
+find it out by typing in your terminal::
     
     ~ $ psql -l
 
 This command will list all of your databases along with their owners. 
-
-``db_cluster_name`` is the name of your Postgres database cluster (if you used
-commands as above your ``Postgres database cluster`` name is `'postgres_dbs'`).
 
 Test
 ----

